packages/discovery-provider/src/utils/auth_middleware.py

```diff
@@ -15,6 +15,37 @@
 SIGNATURE_HEADER = "Encoded-Data-Signature"
 
 
+def recover_authority_from_signature_headers() -> tuple[int | None, str | None]:
```

Comment:
My only regret here is that if you mix `auth_middleware` and code that uses this function deep down at the query level, you end up recovering twice and it's not exactly a speedy operation.